packs/octopusdeploy/actions/lib/actions.py

```diff
@@ -0,0 +1,59 @@
+try:
+    import requests
+except ImportError:
+    message = ('Missing "requests", please install it using pip:\n'
+               'pip install requests')
+    raise ImportError(message)
+
+try:
+    import json
+except ImportError:
+    message = ('Missing "json", please install it using pip:\n'
+               'pip install requests')
+    raise ImportError(message)
+
+from st2actions.runners.pythonrunner import Action
+
+__all__ = [
+    'OctopusDeployAction',
+]
+
+
+class OctopusDeployAction(Action):
+    def __init__(self, config):
+        super(OctopusDeployAction, self).__init__(config)
+        self.client = self._init_client()
+
+    def _init_client(self):
+        api_key = self.config['api_key']
+        host = self.config['host']
+        port = self.config['port']
+        return OctopusDeployClient(api_key=api_key, host=host, port=port)
+
+    def _build_uri(self):
+        # big assumption but it'll cover 99% case,
+        # as octopus runs https by default
+        start = "http://" if self.client.port is 80 else "https://"
```

Comment:
This won't work if a user runs an instance on a non-standard port (80) without SSL, but it's probably not a big deal right now.

I don't know if they even support running it on a non-standard port and SSL should be preferred anyway :)